client/core/rs/src/entities/mod.rs

```diff
@@ -430,10 +430,14 @@ impl CloneArgs {
       .repo
       .as_ref()
       .context("resource has no repo attached")?;
-    Ok(format!(
-      "{protocol}://{access_token_at}{}/{repo}.git",
```

Comment:
I found this discussion in the Gitea repo: https://github.com/go-gitea/gitea/issues/29426

It seems like ending in `.git` is never needed and can be removed here, avoiding any provider specific behavior.

Reply:
Removing `.git` in general case can be tested in pre release out of 1.17.0 branch, if you would like to get the commit in you can make this change and point this PR to 1.17.0 branch, otherwise I will do it over there and close this PR.